### PR TITLE
scons-utils.eclass: EAPI support update, cleanup

### PR DESCRIPTION
**Pull request by mgorny**
- remove support for EAPI 0..6
- add support for EAPI 8
- use `python_has_version`

### CODE REVIEW

**Comment by mgorny** [comment]
Note: scim-sunpinyin still requires EAPI 6 but it's last rited (again).

File: eclass/scons-utils.eclass [hunk 1]
```diff
@@ -88,24 +81,10 @@
 # Much like EXTRA_EMAKE, this is not supposed to be used in make.conf
 # and not in ebuilds!
 
-# @ECLASS_VARIABLE: USE_SCONS_TRUE
```
Comment (thesamesam) [inline eclass/scons-utils.eclass:91]: Oh my.

File: eclass/scons-utils.eclass [hunk 2]
```diff
@@ -88,24 +81,10 @@
...
-# The default value for false in scons-use() (0 by default).
-: ${USE_SCONS_FALSE:=0}
-
 # -- EAPI support check --
 
 case ${EAPI:-0} in
-	0|1|2|3|4|5|6|7) ;;
+	7|8) ;;
 	*) die "EAPI ${EAPI} unsupported."
 esac
```
Comment (thesamesam) [inline eclass/scons-utils.eclass:90]: wrt SCONS_MIN_VERSION: let's just default to 4.4.0 (current stable) but allow override.

That is, we can unconditionally do:
: ${WANT_AUTOMAKE:=4.4.0}
SCONS_DEPEND=">=dev-util/scons-${SCONS_MIN_VERSION}"

It's a lot simpler.

File: eclass/scons-utils.eclass [hunk 3]
```diff
@@ -200,22 +162,7 @@ escons() {
...
-			4|5) # 100% fatal in 4 & 5
-				die "escons failed."
-				;;
-			*) # respect nonfatal in 6 onwards
-				die -n "escons failed."
-				;;
-		esac
-	fi
-	return ${ret}
+	"${@}" || die -n "escons failed."
```
Comment (thesamesam) [inline eclass/scons-utils.eclass:165]: edo

File: eclass/scons-utils.eclass (hunk 2 above)
Comment (mgorny) [reply eclass/scons-utils.eclass:90]: Man, that's a major jump in automake releases xP.

File: eclass/scons-utils.eclass (hunk 3 above)
Comment (mgorny) [reply eclass/scons-utils.eclass:165]: I hate it.

**Comment by mgorny** [comment]
Updated. Also replaced custom `MAKEOPTS` cleaner with `makeopts_jobs`.

**Review by ajakk — APPROVED** [review]

**Pull request closed without merging**